companieshouse/companieshouse-1.0.0/org.wso2.carbon.connector/src/main/resources/company/companieshouselookup.xml

```diff
@@ -0,0 +1,47 @@
+<!--
+   Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+
+   WSO2 Inc. licenses this file to you under the Apache License,
+   Version 2.0 (the "License"); you may not use this file except
+   in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied. See the License for the
+   specific language governing permissions and limitations
+   under the License.
+-->
+<template xmlns="http://ws.apache.org/ns/synapse" name="companieshouselookup">
+
+        <parameter name="company" description="The company being searched for"/>
+        <parameter name="itemsPerPage" description="items per page"/>
+        <parameter name="startIndex" description="start index"/>
+        <sequence>
+            <property name="uri.var.company" expression="$func:company"/>
+            <property name="uri.var.itemsPerPage" expression="$func:itemsPerPage"/>
+            <property name="uri.var.startIndex" expression="$func:startIndex"/>
+            <script language="js"><![CDATA[
+            var itemsPerPage=mc.getProperty('uri.var.itemsPerPage');
+            var startIndex=mc.getProperty('uri.var.startIndex');
+            var query="&";
+            if(itemsPerPage != null && itemsPerPage != ""){
```

Comment:
As an improvement can't we just do the validation if(itemsPerPage) ? You could refer http://stackoverflow.com/questions/5515310/is-there-a-standard-function-to-check-for-null-undefined-or-blank-variables-in for more details